draft-erdtman-jose-cleartext-jws.xml

```diff
@@ -91,36 +91,68 @@
-    </t>
+      The table below shows the primary characteristics of the JWS variants.
+      </t>
+      <texttable anchor="fig:table_jws_variants">
+        <ttcol align="left"></ttcol>
+        <ttcol align="left">JWS</ttcol>
+        <ttcol align="left">Cleartext JWS</ttcol>
+        <c>Data&#xa0;to&#xa0;be&#xa0;Signed</c>
+        <c>Any&#xa0;type and&#xa0;format</c>
+        <c>
+          Non-intrusive signature object
+          for inclusion in arbitrary JSON
+          and JavaScript objects
+        </c>
+        <c>Encoding&#xa0;of&#xa0;Signed&#xa0;Data</c>
+        <c>Base64Url</c>
+        <c>none</c>
+        <c>Encoding of Header Data</c>
+        <c>Base64Url</c>
+        <c>none</c>
+        <c>URL Friendly</c>
+        <c>Core&#xa0;feature</c>
+        <c>Out of scope</c>
+      </texttable>
     <t>
       In the following example, there are a few things to note.
       The signature is included in the data.
       The members in the <spanx style="verb">__cleartext_signature</spanx> object are the JWS
       header parameters.
       The <spanx style="verb">signature</spanx> member contains the base64url encoded signature bytes.
+      </t>
+    <t>
       <figure align="center" anchor="fig:examplesignature" title='Cleartext JWS signed example'>
         <artwork>
-<![CDATA[
-{
-  "iss":"joe",
-  "exp":1300819380,
-  "http://example.com/is_root":true,
+<![CDATA[{
+  "iss": "joe",
+  "exp": 1300819380,
+  "escapeMe": "\u20ac$\u000F\u000aA'\u0042\u0022\u005c\\\"\/",
+  "numbers": [1e+30,4.5,6],
   "__cleartext_signature": {
     "alg": "ES256",
-    "jwk": {
-      "kty": "EC",
-      "crv": "P-256",
-      "x": "_gow8fcS3Dx9z6j57U5q8tunnRBdrgLU9A7CZTYCnqU",
-      "y": "bdfJGraBVL5aPj38TG4tHwxpU2VKwG1XBp0wQfCLOFQ"
-    },
-    "signature": "aRx2MQyCGVOZGViAC_7bEDUp8_CGO1kU1l7Lvp1FHx4qBiPkGs9Z7T
-                  KGK774XLTGwaCfUtd1VrscabQhmArCxA"
+    "kid": "example.com:p256",
+    "signature": "pXP0GFHms0SntctNk1G1pHZfccVYdZkmAJktY_hpMsI
+                  AckzX7wZJIJNlsBzmJ1_7LmKATiW-YHHZjsYdT96JZw"
   }
-}
-]]>
-        </artwork>
+}]]></artwork>
       </figure>
-      <!-- TBD The examples should include the keys used so that they can be reproduced and validated -->
+        </t><t>
+      The following private key in JWK <xref target="RFC7517"/> format can be used for verifying the example signature.
+       </t><t>
+      <figure align="center" anchor="fig:exampleverificationkey" title='Example signature verification key'>
+        <artwork>
+<![CDATA[{
+  "kid": "example.com:p256",
+  "kty": "EC",
+  "crv": "P-256",
+  "x": "censDzcMEkgiePz6DXB7cDuwFemshAFR90UNVQFCg8Q",
+  "y": "xq8rze6ewG0-eVcSF72J77gKiD0IHnzpwHaU7t6nVeY",
+  "d": "nEsftLbi5u9pI8B0-drEjIuJzQgZie3yeqUR3BwWDl4"
+}]]></artwork>
+    </figure>
+         </t><t>
+    Note: Recreating the example signature using the example private key would normally generate a different <spanx style="verb">signature</spanx> value.
```

Comment:
Since random data is added in the signature process? right? I know in COSE we/they opted for the deterministic version of the signature algorithm.
I would like a few words added to make this clear.

Reply:
You are right. JWS is not talking about this and deterministic is not even available in JDK.

How about adding "assuming that you are using a non-deterministic implementation of ES256"